gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/AuthenticateEndpointStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.security.service.zosmf;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;
+import org.zowe.apiml.security.common.error.ServiceNotAccessibleException;
+import org.zowe.apiml.security.common.token.TokenNotValidException;
+
+import static org.zowe.apiml.gateway.security.service.zosmf.AbstractZosmfService.ZOSMF_CSRF_HEADER;
+
+/**
+ * Strategy to validate token through Authentication endpoint of zOSMF
+ */
+@RequiredArgsConstructor
+public class AuthenticateEndpointStrategy implements TokenValidationStrategy {
+

```

Comment:
What are the possibilities for next strategy? Isnt it always against some rest endpoint when it comes to zosmf?

Reply:
I plan to implement a generic authenticated endpoint. The endpoints might have special behaviors so that's why this modularization is selected. In the future, I see the possibility that the tokens might be validated inside apiml if we can have the secrets. This architecture should enable this to be easy to implement.

Reply:
That sounds good. I would consider parametrise the endpoint as it could have the same behaviour and this class then could be reused.